lib/src/DynamicProto.ts

```diff
@@ -133,25 +143,27 @@ function _isObjectArrayOrFunctionPrototype(target:any)
  * @ignore
  */ 
 function _getObjProto(target:any) {
+    let newProto;
+
     if (target) {
         // This method doesn't existing in older browsers (e.g. IE8)
         if (_objGetPrototypeOf) {
             return _objGetPrototypeOf(target);
         }
 
-        // target[Constructor] May break if the constructor has been changed or removed
-        let newProto = target[DynProtoPolyProto] || target[str__Proto] || target[Prototype] || (target[Constructor] ? target[Constructor][Prototype] : null);
-        if(newProto) {
-            // Set the _dynProto so the hierarchy can lookup base classes correctly on IE8
-            if (!target[DynProtoPolyProto]) {
-                target[DynProtoPolyProto] = newProto;
-            }
+        let curProto = target[str__Proto] || target[Prototype] || (target[Constructor] ? target[Constructor][Prototype] : null);
 
-            return newProto;
+        // Using the pre-calculated value as IE8 doesn't support looking up the prototype of a prototype and thus fails for more than 1 base class
+        newProto = target[DynProtoBaseProto] || curProto;
+        if (!_hasOwnProperty(target, DynProtoBaseProto)) {
+            // As this prototype doesn't have this property then this is from an inherited class so newProto is the base to return so save it
+            // so we can look it up value (which for a multiple hierarchy dynamicProto will be the base class)
+            newProto = target[DynProtoBaseProto] = target[DynProtoCurrent] || target[DynProtoBaseProto];
```

Comment:
This might look strange, but the trick here is that because the previous prototype has a `DynProtoCurrent` value of that prototype is what we get (before we assign the current one on the next line).